### PR TITLE
fix: rename samplename in final vcf header

### PR DESCRIPTION
**Pull request by mathiasbio**
### This PR:

This PR aims to change the name of the sample columns in the final VCFs to match the changes in the scout_load.yaml for sample-id, which will use LIMS-ID instead of TUMOR/NORMAL. (https://github.com/Clinical-Genomics/cg/pull/2650)

Added:
- New rule to create namemap to translate sample type names in vcf header to sampleID

Changed: 
- Renamed names of sample columns in final vcf (clinical.pass.vcf.gz) to sampleID

Removed:
- Removed sed command in CNVpytor rule introduced here (https://github.com/Clinical-Genomics/BALSAMIC/pull/1182) as it has been fixed by updating CNVpytor and is no longer necessary (https://github.com/Clinical-Genomics/BALSAMIC/issues/1152)

### PR specific tests
- [x] TGA T/N case finishes without error
- [x] And both clinical.pass vcfs have the changed sample-columns names
- [x] TGA UMI case finishes without error
- [x] And both clinical.pass vcfs have the changed sample-columns names
- [x] WGS T/N case finishes without error
- [x] And both clinical.pass vcfs have the changed sample-columns names
- [x] WGS T-only case finishes without errors
- [x] And both clinical.pass vcfs have the changed sample-columns names

### Review and tests: 
- [x] Tests pass
- [x] Code review
- [x] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: BALSAMIC/snakemake_rules/annotation/final_vcf_reheader.rule [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# vim: syntax=python tabstop=4 expandtab
+# coding: utf-8
+
+rule create_final_vcf_namemap:
+    input:
+        multiqc_json = qc_dir + "multiqc_data/multiqc_data.json",
+    output:
+        namemap = vep_dir + "status_to_sampleid_namemap"
+    params:
+        status_to_sampleid = status_to_sampleid
```
Comment (ivadym) [inline BALSAMIC/snakemake_rules/annotation/final_vcf_reheader.rule:10]:
```suggestion
        status_to_sample_id = status_to_sample_id
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_normal.rule [hunk 2]
```diff
@@ -156,7 +157,8 @@ bcftools view {input.vcf_snv_clinical} | \
 bcftools filter --threads {threads} --include 'INFO/GNOMADAF_popmax <= {params.pop_freq[0]} || INFO/GNOMADAF_popmax == \".\"' --soft-filter '{params.pop_freq[1]}' --mode '+' | \
 bcftools filter --threads {threads} --include 'INFO/SWEGENAF <= {params.swegen_freq[0]} || INFO/SWEGENAF == \".\"' --soft-filter '{params.swegen_freq[1]}' --mode '+' |\
 bcftools filter --threads {threads} --include 'INFO/Frq <= {params.loqusdb_clinical_freq[0]} || INFO/Frq == \".\"' --soft-filter '{params.loqusdb_clinical_freq[1]}' --mode '+' |\
-bcftools view --threads {threads} -f PASS -o {output.vcf_pass_vardict}.temp1 -O z;
+bcftools view --threads {threads} -f PASS |\
+bcftools reheader -s {input.namemap} -o {output.vcf_pass_vardict}.temp1;
```
Comment (ivadym) [inline BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_normal.rule]: We've removed the `-O z`, does this mean that we are not compressing the output anymore? Is this intentional?

File: BALSAMIC/snakemake_rules/annotation/vcfheader_rename.rule [hunk 3]
```diff
@@ -29,4 +29,4 @@ bcftools reheader --threads {threads} \
 -o {output.vcf_dnascope};
 
 tabix -p vcf -f {output.vcf_dnascope}
-        """
+        """
```
Comment (ivadym) [inline BALSAMIC/snakemake_rules/annotation/vcfheader_rename.rule:32]: EOF

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_only.rule [hunk 4]
```diff
@@ -243,8 +243,6 @@ cp {params.tmpdir}/{params.tumor}.global.0000.png {output.scatter_cnvpytor};
 
 cp {params.tmpdir}/{params.tumor}.circular.0001.png {output.circular_cnvpytor};
```
Comment (ivadym) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_only.rule:245]: Could you include this in the CHANGELOG?

File: BALSAMIC/workflows/balsamic.smk [hunk 5]
```diff
@@ -94,6 +94,12 @@ sequencing_type = config_model.analysis.sequencing_type
 if config_model.analysis.analysis_type == "paired":
     normal_sample: str = config_model.get_sample_name_by_type(SampleType.NORMAL)
 
+# Sample status to sampleID namemap
+if config_model.analysis.analysis_type == "paired":
+    status_to_sampleid = "TUMOR" + "\\\\t" + tumor_sample + "\\\\n" + "NORMAL" + "\\\\t" + normal_sample
```
Comment (ivadym) [inline BALSAMIC/workflows/balsamic.smk:99]:
```suggestion
    status_to_sample_id = "TUMOR" + "\\\\t" + tumor_sample + "\\\\n" + "NORMAL" + "\\\\t" + normal_sample
```

**Review by ivadym — COMMENTED** [review]
Looks good 👍  Left you some minor comments

**Review by ivadym — APPROVED** [review]
Look great after your last push! 🥇 🌟 🏎️ 💨 🌟 🥇

File: BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_normal.rule (hunk 2 above)
Comment (mathiasbio) [reply BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_normal.rule]: Outdated sorry :D we're compressing them!

**Pull request merged**